src/core/tests/Test_PatchAssessor.py

```diff
@@ -66,6 +68,85 @@ def test_assessment_telemetry_fail(self):
         self.assertRaises(Exception, self.runtime.patch_assessor.start_assessment)
         self.runtime.patch_assessor.telemetry_writer = backup_telemetry_writer
 
+    def test_assessment_state_file(self):
+        # read_assessment_state creates a vanilla assessment state file if none exists
+        assessment_state = self.runtime.patch_assessor.read_assessment_state()
+        with open(self.runtime.patch_assessor.assessment_state_file_path, 'r') as file_handle:
+            assessment_state_from_file = json.loads(file_handle.read())["assessmentState"]
+            self.assessment_state_equals(assessment_state, assessment_state_from_file)
+
+        # write and test again
+        self.runtime.patch_assessor.write_assessment_state()
+        assessment_state = self.runtime.patch_assessor.read_assessment_state()
+        with open(self.runtime.patch_assessor.assessment_state_file_path, 'r') as file_handle:
+            assessment_state_from_file = json.loads(file_handle.read())["assessmentState"]
+            self.assessment_state_equals(assessment_state, assessment_state_from_file)
```

Comment:
Another way to validate write_assessment_state(), updated an existing AssessmentState.json will be to compare the lastHeartbeat before and after write_assessment_state(), i.e:

- Read AssessmentState before line 79 i.e. before write_assessment_state()
- Call write_assessment_state()
- Read the file
- compare lastHeartbeat between step 1 and 3, should not be the same

Reply:
Ben is aware of what to do here. Ack